packages/vite/src/node/utils.ts

```diff
@@ -573,20 +574,42 @@ export async function processSrcSet(
     })
     .filter(({ url }) => !!url)
 
-  const ret = await Promise.all(
-    imageCandidates.map(async ({ url, descriptor }) => {
-      return {
-        url: await replacer({ url, descriptor }),
+  const formatSrcset = (ret: { url: string; descriptor: string }[]) =>
+    ret.reduce((prev, { url, descriptor }, index) => {
+      descriptor ??= ''
+      return (prev +=
+        url + ` ${descriptor}${index === ret.length - 1 ? '' : ', '}`)
+    }, '')
+
+  if (isSync) {
```

Comment:
I think it is better to have two functions: `replaceSrcSet` and `replaceSrcSetSync`, and you can extract some common pieces if needed, or repeat some code.

Reply:
The problem confused me. I think now it seems good. 👍

Reply:
A lot better now, thanks for making the change!